### PR TITLE
Rewrite preferences code. Make new defaults.

### LINKED ISSUE #3259: Four (or so) of the *default* values for local preferences are "backwards"
Every time I teach a new player, the first thing we have to do is go into their gear icon to turn on a few settings:

- Learner mode
- Enable sounds
- Magnify cards on hover
- Show in-game alerts

Should these be the defaults for new players?

### PR DESCRIPTION
**Pull request by kberg**
Enables the following preferences as defaults:
* Learner mode
* Enable sounds
* Magnify cards on hover
* Show in-game alerts

The rewrite is probably a little more code, but it relies more on
natural typing and simpler. Though it could use a little more work. I'd
like to remove PreferencesManager.INSTANCE.values() everywhere.

This fixes #3259

### CODE REVIEW

**Comment by kberg** [comment]
@bafolts can you take a look at this please? I need to add more tests, but I really want to know what you think of this change.

File: src/client/components/PreferencesDialog.vue [hunk 1]
```diff
@@ -42,52 +30,25 @@ export default Vue.extend({
...
-          } else {
-            strVal = this.$data[k] ? '1' : '0';
-          }
-          PreferencesManager.save(k, strVal);
-          PreferencesManager.preferencesValues.set(k, this.$data[k]);
-          this.setPreferencesCSS(this.$data[k], k);
-        }
+    updatePreferences() {
+      for (const k of Object.keys(this.preferencesManager.values) as Array<Preference>) {
+        const val = this[k];
```
Comment (bafolts) [inline src/client/components/PreferencesDialog.vue:35]:
```suggestion
        const val = this.preferencesManager.values[k];
```

As written this is assuming that `this.data` is set to `this.preferencesManager.values()`.

Comment (kberg) [reply src/client/components/PreferencesDialog.vue:35]: That's right. If something changes this, well, the whole thing would need to change a little bit.

**Comment by kberg** [comment]
There are now tests, and I moved alot around -- again. But it's good now.

File: src/client/components/PlayerHome.vue [hunk 2]
```diff
@@ -283,22 +283,23 @@ class TerraformedAlertDialog {
...
+      showActiveCards: !preferences.hide_active_cards,
+      showAutomatedCards: !preferences.hide_automated_cards,
+      showEventCards: !preferences.hide_event_cards,
       hideTiles: false,
     };
   },
   watch: {
     hide_active_cards() {
-      PreferencesManager.save('hide_active_cards', !this.showActiveCards);
+      PreferencesManager.INSTANCE.set('hide_active_cards', !this.showActiveCards);
```
Comment (bafolts) [inline src/client/components/PlayerHome.vue:296]: There is a getPreferences now. A setPreference may work here.

**Review by bafolts — APPROVED** [review]

**Pull request merged**